### ISSUE TITLE
Enhancement: Stop current audio playback when new audio is triggered

### ISSUE THREAD

**Issue by denete** [FrSkyRC/ETHOS-Feedback-Community#4456]
I would like to suggest that playing a track or value should be stopped if another event triggers another track or value to be played.

**Comment by bsongis-frsky** [comment]
This one is difficult, would you like that the "RSSI low" is cut just because there is a "Landing gear out" track to play?

**Comment by Wolepo** [comment]
Here again, a categorisation into important and unimportant would come into play.

**Comment by pstasek** [comment]
Agree... at the same time I would want to hear RSSI Low when it happens, not after a Landing Gear Out track or a whole queue of tracks finish to play. Maybe some kind of priority system? Or a "Can be cut" flag on each track? 

I can imagine that notifications such as gear down, gear up, flaps up, flaps 1, flaps 2 can be cut but for alarms such as battery low/rssi or vfr I'd like to hear these uncut.

Freestyle acrobatic pilots would like to trigger a music track to start the sequence, for example. These would block alarms at all.

**Comment by craigywsm** [comment]
maybe system announcements e.g. RSSI / Radio battery low etc should always be top priority.
If announcement (list of) are playing it pauses them to then give RSSI and then resumes

**Comment by Wolepo** [comment]
And if you move the warning messages from the system to a separate audio folder and give them their own audio track, as with the Vario? If you add an additional checkbox [important] in the SF ‘Play Audio’ and add these announcements to the new folder, you could decide for yourself what is also considered important or not.

**Comment by strgaltdel** [comment]
Wolfgang, your proposal was my 1st thought too,

However, we should remember that the new audio structure in version 1.5 was a hurdle for many users and generated quite a bit of confusion in the forum. Any further changes need to be carefully considered. In my opinion, they should be implemented without requiring significant user intervention.

If it were possible to migrate:

Warnings to /system/warnings
Alarms to /system/alarms

This could establish a (descending) priority:
- Alarms
- Warnings
- Infos

This priority would be easily visible to the user based on the folder assignment. However, the priority would still need to be coded within Ethos itself.

After that, there might be a discussion about how to handle user-specific files. I can imagine that some users might copy their audio files into /warnings and expect a corresponding "dynamic" assignment.

This would add another layer of complexity. Is it really necessary?

regards
udo

**Comment by Wolepo** [comment]
Of course, it is true that you have to reckon with the user's ignorance. But at the moment, for example, voltage warnings are not provided for in .../System and are only issued after the end of a running announcement, unless there are others in the queue. For me, it would be good if these warnings were simply output at the same time as the running audio file. The icing on the cake would be to adjust the volume accordingly.
Regards 
Wolfgang

**Comment by robthomson** [comment]
I think this could be a big issue.

For example in lua its commong to use multiple wav files.

So

wav file 1:   governor.wav
wav file 2:  active.wav
wav file 3:  disabled.wav

You then 'chain them' to play things like:   "governor active"

**Comment by philippe-warnotte** [comment]
Perfect is the enemy of good !

**Comment by bsongis** [comment]
@philippe-warnotte Your translation is not right!

**Comment by philippe-warnotte** [comment]
Sorry

**Comment by bsongis-frsky** [comment]
I can only think of adding some priority to each prompt. Only a prompt with an higher priority may cut the current one. But even with this in mind, it is easy to do something very bad!

**Comment by loetefix** [comment]
I think the following is better than a priority.

For this it would be useful to have a Group number
option to voice call outs.

voice call outs with the same number can interrupt an other with same groupnumber but not call outs with different numbers.

For example you attach number 1 to flightmode and number 2 to valtage call outs, flightmodes will not interrupt a voltage call out . a flightmode call out will interrupt an other flightmode callout with same "ID" (1) in this case. And group 0 will never interrupted (reseved for alarms).

I think this would be a good solution.

**Comment by bsongis-frsky** [comment]
This is exactly what I would call a priority!

**Comment by 3Dholic** [comment]
Is there any update about the subject ?

**Comment by robthomson** [comment]
I can't sew personally how this would work.

Single sounds are chained together.


One.wav
Hundred.wav

"one hundred"

Net result of stopping audio would be getting

"hundred"

**Comment by craigywsm** [comment]
Not Stale
